lib/block_keys/ckd.ex

```diff
@@ -116,35 +118,42 @@ defmodule BlockKeys.CKD do
     )
   end
 
-  def master_public_key(<<"xpub", _rest::binary>>),
+  def master_public_key(key, opts \\ [])
+
+  def master_public_key(<<"xpub", _rest::binary>>, _opts),
     do: {:error, "Cannot derive master public key from another extended public key"}
 
-  def master_public_key(<<"tpub", _rest::binary>>),
+  def master_public_key(<<"tpub", _rest::binary>>, _opts),
     do: {:error, "Cannot derive master public key from another extended public key"}
 
-  def master_public_key(key) do
+  def master_public_key(key, opts) do
     decoded_key = Encoding.decode_extended_key(key)
 
     data =
       decoded_key
       |> slice_prefix()
-      |> put_uncompressed_parent_pub(%{index: decoded_key.index})
-      |> put_compressed_parent_pub()
+      |> put_parent_pub(%{index: decoded_key.index}, opts)
 
-    network =
+    {network, prefix} =
       case key do
-        "xprv" <> _ -> :mainnet
-        "tprv" <> _ -> :testnet
+        "xprv" <> _ -> {:mainnet, "xpub"}
+        "tprv" <> _ -> {:testnet, "tpub"}
       end
 
-    Encoding.encode_extended_key(
-      Encoding.public_version_number(network),
-      decoded_key.depth,
-      decoded_key.fingerprint,
-      decoded_key.index,
-      decoded_key.chain_code,
-      data.parent_pub_key
-    )
+    encoded_public_key =
+      Encoding.encode_extended_key(
+        Encoding.public_version_number(network),
+        decoded_key.depth,
+        decoded_key.fingerprint,
+        decoded_key.index,
+        decoded_key.chain_code,
+        data.parent_pub_key
+      )
+
+    case opts[:network] do
+      :solana -> prefix <> encoded_public_key
```

Comment:
maybe add a reference here on why a prefix is required for Solana?

Reply:
Hi @wchenNL, there's no real reference if you mean a reference doc:) Just during encoding, it can't decode hex first [here](https://github.com/ExWeb3/block_keys/blob/master/lib/block_keys/base58/check.ex#L25) which returns the encoded key without `xpub` prefix. Added the prefix to keep the format of key which is later pattern matched in `from_xpub`.